src/commands/plugins.rs

```diff
@@ -0,0 +1,275 @@
+use anyhow::{anyhow, Result};
+use clap::{Parser, Subcommand};
+use semver::Version;
+use spin_plugins::{
+    install::{ManifestLocation, PluginInfo, PluginInstaller},
+    uninstall::PluginUninstaller,
+    PLUGIN_MANIFESTS_DIRECTORY_NAME,
+};
+use std::path::PathBuf;
+use tracing::log;
+use url::Url;
+
+const SPIN_PLUGINS_REPO: &str = "https://github.com/fermyon/spin-plugins/";
+
+/// Install/uninstall Spin plugins.
+#[derive(Subcommand, Debug)]
+pub enum PluginCommands {
+    /// Install plugin from a manifest.
+    ///
+    /// The binary file and manifest of the plugin is copied to the local Spin
+    /// plugins directory.
+    Install(Install),
+
+    /// Remove a plugin from your installation.
+    Uninstall(Uninstall),
+
+    /// Upgrade one or all plugins.
+    Upgrade(Upgrade),
+}
+
+impl PluginCommands {
+    pub async fn run(self) -> Result<()> {
+        match self {
+            PluginCommands::Install(cmd) => cmd.run().await,
+            PluginCommands::Uninstall(cmd) => cmd.run().await,
+            PluginCommands::Upgrade(cmd) => cmd.run().await,
+        }
+    }
+}
+
+/// Install plugins from remote source
+#[derive(Parser, Debug)]
+pub struct Install {
+    /// Name of Spin plugin.
+    #[clap(
+        name = "PLUGIN_NAME",
+        conflicts_with = "REMOTE_PLUGIN_MANIFEST",
```

Comment:
Use consts for argument names; that way, if a name changes, it doesn't have to be kept in sync across multiple places.